### DISCUSSION TITLE
Odd behavior feeding kitty @ get-colors output to fzf

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#3770] [Q&A] [answered]
I am trying to feed the output of `kitty @ get-colors` to `fzf`, like this:

```
kitty @ get-colors | fzf
```

This does not work.  `fzf` hangs and I see garbage on the screen.  I have to `CTRL+C` to kill `fzf`.

On the other hand, if I split this command into two separate pipes, then it works:

```
colors="$(kitty @ get-colors)"
echo "$colors" | fzf
```

1. What is different about the way Kitty generates the output that is failing `fzf`?
2. Is this something that that should be fixed in Kitty, or shall I file a bug on `fzf`?

**Comment by kovidgoyal** [comment, marked as answer]
Use a socket for remote control, otherwise both kitty and fzf will be doing I/O to the terminal simultaneously. Or  route the output from remote control somewhere and after it quits run fzt.
